solo-io/solo-apis
# Support Multiple LTS Branches

# Description

Introduce GitHub Actions that support the automated tracking of LTS branches, used for Gloo Edge.

# Context

See https://github.com/solo-io/gloo/pull/4999 for an overview of why this change helpful.

## How do these workflows function?
I added a README to outline the purpose of the workflows, and some of the design considerations.

## Why are they net new?
There are existing workflows that automate the process of merging API syncs into master. Instead of modifying them, I decided to create net new workflows, for a couple of reasons.

#### They work, and while these new ones are tested, they haven't been proven to behave correctly.
The existing workflows are useful for developer efficiency. Breaking them would slow down team members, which I'd like to avoid.

#### These workflows operate on different triggers
These workflows respond to different triggers (ie different branch names) and therefore it's ok to have them both exist at the same time. It means that we can do a bit of a Canary rollout for these workflows:
1. Add both in solo-apis
2. Support both in gloo
3. Run a release
4. Monitor the effect and compare the generated branches

#### To simplify the git history
Similar to the first reason, in case something goes wrong, I want it to be easy to delete these actions.
Also, if they are great, it will be easy to backport these changes to other LTS branches already existing in gloo-edge

## How can I test this?
- Push some commit to a branch named `sync-apis/some-lts-branch-name/some-tag-name`. Ensure that if `some-lts-branch-name` branch doesn't exist, it is created, and a PR is opened against it. Also, ensure that a commit is added with a message that includes `@tag-name=some-tag-name`
